dvc/cli.py

```diff
@@ -108,6 +142,13 @@ def error(self, message, cmd_cls=None):  # pylint: disable=arguments-differ
         _find_parser(self, cmd_cls)
 
     def parse_args(self, args=None, namespace=None):
+        if args is None:
+            args = sys.argv[1:]
+        else:
+            args = list(args)
```

Comment:
Why do we need to do this during normal operation?

Reply:
Based on this following comment from Peter https://github.com/iterative/dvc/pull/4483#discussion_r479999776
> When DVC is run from the command line, args will be None here (meaning argparse should default to using sys.argv), so the suggested command code is never hit (except when we explicitly call parse_args with an argument list from the unit tests).

Reply:
Without those lines, we get the following error:
```python
ERROR: unexpected error - object of type 'NoneType' has no len()
```

Reply:
@santos22 we didn't need that before, so probably caused by changes in this PR. argparse takes it from somewhere without this change, so I wonder why we need this.

Reply:
Right. These are the internal `argparse` methods we are using:
```python
def parse_args(self, args=None, namespace=None):
    args, argv = self.parse_known_args(args, namespace)
    if argv:
        msg = _('unrecognized arguments: %s')
        self.error(msg % ' '.join(argv))
    return args

def parse_known_args(self, args=None, namespace=None):
    if args is None:
        # args default to the system args
        args = _sys.argv[1:]
    else:
        # make sure that args are mutable
        args = list(args)
    ...
```
[Source](https://github.com/python/cpython/blob/b084d1b97e369293d2d2bc0791e2135822c923a8/Lib/argparse.py#L1817)

It appears `parse_known_args()` defaults to the system args for us. This method is the one that throws the `argument cmd: invalid choice` error which we aim to override with a command suggestions error.

Because we are working with the `args` parameter outside of this method, we have to use those very same lines.